libs/common/src/test/java/org/opensearch/common/round/RoundableTests.java

```diff
@@ -8,19 +8,53 @@
 
 package org.opensearch.common.round;
 
+import org.opensearch.common.SuppressForbidden;
 import org.opensearch.test.OpenSearchTestCase;
+import org.junit.BeforeClass;
 
 public class RoundableTests extends OpenSearchTestCase {
 
-    public void testFloor() {
-        int size = randomIntBetween(1, 256);
-        long[] values = new long[size];
-        for (int i = 1; i < values.length; i++) {
-            values[i] = values[i - 1] + (randomNonNegativeLong() % 200) + 1;
+    @BeforeClass
+    @SuppressForbidden(reason = "Sets the feature flag system property to a random value")
```

Comment:
Please use parametrized test to verify the behaviour w/o system property set. Also please make sure the old value is restored - we run tests in parallel and changing system property in JVM may easily impact all other tests that will be run on it.

```
@Before
public void setUp() {
    oldValue = System.getProperty("opensearch.experimental.feature.simd.rounding.enabled");
    System.setProperty("opensearch.experimental.feature.simd.rounding.enabled", ...);
}

@After
public void tearDown() {
 if (old != null) {
        System.setProperty("opensearch.experimental.feature.simd.rounding.enabled", old);
 } else {
        System.clearProperty("opensearch.experimental.feature.simd.rounding.enabled");
 }
}
```

Reply:
I see another problem: `IS_BTREE_SEARCH_SUPPORTED` is statically initialized so it's value is loaded the first time the class is accessed. Changing the system property before/after each individual test won't work as the static final field is already set. Unless I perform this check inside the `create` method every time we create a Roundable.

Do you have a better way to handle this? Modifying the system property and/or static fields in unit-tests already feels like a code smell to me. 🫣

Reply:
The other option is to make this field non-static package-private and modify in unit-tests. Seems like the simplest and most comprehensible solution.

Reply:
> The other option is to make this field non-static package-private and modify in unit-tests. Seems like the simplest and most comprehensible solution.

It will be costly at runtime (to check non-final field every time when rounding is created). The way we approached such tests in the past  is by having dedicated test task with system property set and tests classes included:

```
task roundableSimdTest(type: Test) {
    include '**/RoundableTests.class'
    systemProperty "opensearch.experimental.feature.simd.rounding.enabled", "true"
}
```

The regular `test` task will run `RoundableTests` without system property set, so we should be able to cover both flows.

Reply:
Great suggestion! This works perfectly for my use-case. Thanks a lot!